### ISSUE TITLE
Unable to run factbeat on ubuntu 12.x

### ISSUE THREAD

**Issue by jpsandiego** [jarpy/factbeat#2]
Here's what i've done on Ubuntu 12.x with fresh docker/docker-compose

* Add 'elasticsearch' to /etc/hosts for the target ES cluster I want to use
* git clone ...
* cd factbeat
* Populate the template into the cluster
* make
* make unit-test
* make acceptance-test

And that fails with these errors a couple of times:
`E           FactbeatError: Factbeat is not populating Elasticsearch.
`

Even going around the acceptance tests, I'm finding I can't get very far:

```
[...~/src/factbeat]$ ./factbeat -c factbeat.yml -e -v -d "*"
2016/03/04 17:52:29.565697 beat.go:221: DBG  Initializing output plugins
2016/03/04 17:52:29.565777 geolite.go:24: INFO GeoIP disabled: No paths were set under shipper.geoip.paths
2016/03/04 17:52:29.565876 client.go:431: DBG  ES Ping(url=http://elasticsearch:9200, timeout=1m30s)
2016/03/04 17:52:29.570664 client.go:440: DBG  Ping status code: 200
2016/03/04 17:52:29.570701 outputs.go:89: INFO Activated elasticsearch as output plugin.
2016/03/04 17:52:29.570717 publish.go:235: DBG  Create output worker
2016/03/04 17:52:29.570818 publish.go:277: DBG  No output is defined to store the topology. The server fields might not be filled.
2016/03/04 17:52:29.570876 publish.go:291: INFO Publisher name: myhost
2016/03/04 17:52:29.571158 async.go:70: INFO Flush Interval set to: 1s
2016/03/04 17:52:29.571195 async.go:71: INFO Max Bulk Size set to: 50
2016/03/04 17:52:29.571211 async.go:79: DBG  create bulk processing worker (interval=1s, bulk size=50)
2016/03/04 17:52:29.571272 beat.go:238: INFO Init Beat: factbeat; Version: 0.1.2
2016/03/04 17:52:29.571544 beat.go:267: INFO factbeat sucessfully setup. Start running.
```

Can you validate the current master runs for you or give me some hints?

**Comment by jarpy** [comment]
Whoa, 10 days with no response, sorry! I had notifications disabled, like a fool.

I'll check this out.

**Comment by jarpy** [comment]
Is that Ubuntu 12.4 or 12.10 you're using? Thanks.

**Comment by jarpy** [comment]
I couldn't replicate the failure on Ubuntu 12.4. In truth, the OS version shouldn't matter much because everything is running in containers.

I may, however, have figured out a silly mistake I made. Are you modifying `factbeat.yml` in place? The tests (stupidly) depend on it. That is now fixed.

Also, from IRC history, it looks like you are building Factbeat from source in order to hack the path to the Facter binary. I've made that configurable and updated the README to lead people to the ready-made binaries (if you _want_ to build it, that's great, if you _need_ to build it, I've failed).

The README is also updated to recommend against Facter version 2, which I think you might be using. It should work, but the structure of the data is not as well defined as version 3. I would you love your feedback on how well it works for you (when we get it working).

Anyway, if you have time, please try [Factbeat v0.1.3](https://github.com/jarpy/factbeat/releases/tag/v0.1.3). Thanks.

**Comment by jarpy** [comment]
Possibly interesting:

`factbeat -d '*'` suppresses the error message about a missing Facter binary.

```
$ ./factbeat 
2016/03/15 13:37:59 fork/exec /opt/puppetlabs/bin/facter: no such file or directory

$ ./factbeat -d '*'

$
```

@andrewkroh: I think I'm going mad!

**Comment by jarpy** [comment]
OK. Switched to using Beats-style logging. Much better now.

Sorry @jpsandiego, if I'd done this from the start, I suspect you would have found your problem a lot sooner.

**Comment by jarpy** [comment]
Logging is fixed in [v0.1.4](https://github.com/jarpy/factbeat/releases/tag/v0.1.4).
